### PR TITLE
Use inquirer's 'when' instead of 'filter' to check if a package name is available on npm or Bower

### PR DESCRIPTION
**Pull request by zenorocha**

### CODE REVIEW

**Comment by sindresorhus** [comment]
ping @zenorocha

**Comment by zenorocha** [comment]
Rebased, sorry for the delay.

File: app/index.js [hunk 1]
```diff
@@ -27,32 +24,50 @@ module.exports = yeoman.generators.Base.extend({
...
       name: 'name',
       message: 'Project Name',
       default: this.appname,
-      filter: function (input) {
+    }, {
+      type: 'confirm',
+      name: 'pkgName',
+      message: 'The name above already exists on npm or Bower, choose another?',
+      default: true,
+      when: function(answers) {
```
Comment (kevva) [inline app/index.js]: `function (answers)`

File: app/index.js [hunk 2]
```diff
@@ -10,10 +10,7 @@ module.exports = yeoman.generators.Base.extend({
     this.compareVersion = compareVersion;
   },
 
-  prompting: function () {
-    var cb = this.async();
-    var log = this.log;
-
+  welcome: function() {
```
Comment (kevva) [inline app/index.js]: `function ()`

**Comment by zenorocha** [comment]
Updated, thanks @kevva!

**Comment by kevva** [comment]
lgtm!

**Pull request merged**